src/Data/Unfoldable.purs

```diff
@@ -88,6 +89,11 @@ none = unfoldr (const Nothing) unit
 singleton :: forall f a. Unfoldable f => a -> f a
 singleton = replicate 1
 
+-- | Create an Unfoldable containing a range of values, with both endpoints.
+range :: forall f a. Unfoldable f => Ord a => Semiring a => a -> a -> f a
```

Comment:
I'm a little wary that `Ord a => Semiring a => ...` is a bit too general to be useful, without some laws relating them. Why not just use `Int`?

Reply:
done.